File: backend/src/main/java/br/unitins/tp1/pizzaria/model/Pizza.java
package br.unitins.tp1.pizzaria.model;

import jakarta.persistence.*;

import java.util.HashSet;
import java.util.Set;


@Entity
@DiscriminatorValue("PIZZA")
public class Pizza extends Produto {

    @Enumerated(EnumType.STRING)
    private TamanhoPizza tamanhoPizza;

    private Integer quantPorcoes;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<PorcaoPizza> porcoes = new HashSet<>();

    private Boolean pizzaPronta;

    public TamanhoPizza getTamanhoPizza() {
        return tamanhoPizza;
    }

    public void setTamanhoPizza(TamanhoPizza tamanhoPizza) {
        this.tamanhoPizza = tamanhoPizza;
    }

    public Integer getQuantPorcoes() {
        return quantPorcoes;
    }

    public void setQuantPorcoes(Integer quantPorcoes) {
        this.quantPorcoes = quantPorcoes;
    }

    public Set<PorcaoPizza> getPorcoes() {
        return porcoes;
    }

    public void setPorcoes(Set<PorcaoPizza> porcoes) {
        this.porcoes = porcoes;
    }

    public Boolean getPizzaPronta() {
        return pizzaPronta;
    }

    public void setPizzaPronta(Boolean pizzaPronta) {
        this.pizzaPronta = pizzaPronta;
    }
}

File: frontend/src/app/models/pizza.ts
import {Produto} from "./produto";
import {PorcaoPizza} from "./porcao-pizza";

export interface Pizza extends Produto {
  tamanhoPizza: TamanhoPizza;
  quantPorcoes: number;
  porcoes: PorcaoPizza[];
  pizzaPronta: boolean;
}

export enum TamanhoPizza {
  PEQUENA,
  MEDIA,
  GRANDE,
  GIGA
}

export function formatTamanhoPizza(tamanhoPizza: TamanhoPizza): string {
  switch (tamanhoPizza.toString()){
    case "PEQUENA":
      return "Pequena"
    case "MEDIA":
      return "Média"
    case "GRANDE":
      return "Grande"
    case "GIGA":
      return "Giga"
  }
  return ""
}
